### DISCUSSION TITLE
嘿，windows10 开发环境，刚git下来，试了提示未知错误

### DISCUSSION THREAD

**Discussion by aphantom** [halo-dev/halo#4230] [Q&A]
### 是什么版本出现了此问题？

"name": "@halo-dev/console",   "version": "2.7.0"

### 使用的什么数据库？

PostgreSQL

### 使用的哪种方式部署？

Fat Jar

### 发生了什么？

#第一次安装
问题路径：http://localhost:8090/console/  和  http://localhost:8090/console/setup

系统：Windows 10 专业版  22H2   内部版本号：19045.3208

java version "20.0.1" 2023-04-18
Java(TM) SE Runtime Environment (build 20.0.1+9-29)
Java HotSpot(TM) 64-Bit Server VM (build 20.0.1+9-29, mixed mode, sharing)

pnpm -v 8.5.1
node.js  18.14.2

  [#运行如下面提示]
                                                                                                             07:57:10
  VITE v4.0.4  ready in 26993 ms

  ➜  Local:   http://localhost:8090/console/                                                                  07:57:10
  ➜  Network: http://10.37.114.239:8090/console/                                                              07:57:10
  ➜  Network: http://172.23.224.1:8090/console/                                                               07:57:10
  ➜  press h to show help                                                                                     07:57:10
[vite-plugin-static-copy] Collected 9 items.                                                                  07:57:10

 WARN                                                                                                         07:58:12
A PostCSS plugin did not pass the `from` option to `postcss.parse`. This may cause imported assets to be incorrectly transformed. If you've recently added a PostCSS plugin that raised this warning, please contact the package author to fix the issue.

### 附加信息

![image](https://github.com/halo-dev/halo/assets/16313864/2dd05786-352a-4b47-89f7-8d9b0d07cfe3)
![image](https://github.com/halo-dev/halo/assets/16313864/329ebf78-7e7f-42ca-8ab9-80ea2f3ca9ec)
无法登录
![image](https://github.com/halo-dev/halo/assets/16313864/f2410a23-be65-4aeb-bd27-bdf653f44d39)
初始化状态始终不变

**Comment by ruibaby** [comment]
建议参考文档：https://docs.halo.run/developer-guide/core/run

在不熟悉项目的情况下，建议不做任何修改，完全按照文档的步骤进行。

此问题不属于 issue，我将转移到 Discussion。

**Comment by aphantom** [comment]
好吧我删了。再按开发者环境中说的再操作一遍试试。

**Comment by aphantom** [comment]
请问一下我有JDK20.0.1 环境还需不需要安装OpenJDK 17 LTS？

**Reply by guqing** [reply]
没有测试过在 JDK 20 下的情况，可能会有未知问题，可以在 IDEA 的项目配置选择 JDK 版本处直接 download 很方便不需要任何配置
<img width="873" alt="image" src="https://github.com/halo-dev/halo/assets/38999863/301955ca-6694-4332-b5ee-89bfd88f00a1">

**Comment by aphantom** [comment]
很不巧，我用的VSCode Insiders 没有安装IntelliJ，不是很习惯。

**Reply by guqing** [reply]
那你可以现在你的 20 上试试，如果有问题再切换，可以试试 SDK Man : https://sdkman.io/install

**Comment by aphantom** [comment]
我其实的想通过win10直接配置运行环境比如JDK、PostgreSQL、nodejs等等，不安装Docker这类容器，直接用npm或pnpm运行开发模式可以吗？因为我电脑上安装了这些环境，不想再安装其他的软件。最小化运行

**Reply by guqing** [reply]
只要有 JDK 17 和 pnpm 即可运行

**Comment by aphantom** [comment]
JDK换成17，重新安装：
![image](https://github.com/halo-dev/halo/assets/16313864/f261dfa2-f5f8-4e5c-b6e8-957ba608c961)

![image](https://github.com/halo-dev/halo/assets/16313864/857786a7-228c-4c50-90c5-7f4197d7f883)

**Comment by aphantom** [comment]
一切换安文档操作的
PS E:\use> java -version
java version "17.0.7" 2023-04-18 LTS
Java(TM) SE Runtime Environment (build 17.0.7+8-LTS-224)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.7+8-LTS-224, mixed mode, sharing)
PS E:\use> pnpm -ver
Version 8.5.1
Usage: pnpm [command] [flags]
       pnpm [ -h | --help | -v | --version ]

Manage your dependencies:
      add                  Installs a package and any packages that it depends on. By default, any new package is
                           installed as a prod dependency
      import               Generates a pnpm-lock.yaml from an npm package-lock.json (or npm-shrinkwrap.json) file
   i, install              Install all dependencies for a project
  it, install-test         Runs a pnpm install followed immediately by a pnpm test
  ln, link                 Connect the local project to another one
      prune                Removes extraneous packages
  rb, rebuild              Rebuild a package
  rm, remove               Removes packages from node_modules and from the project's package.json
      unlink               Unlinks a package. Like yarn unlink but pnpm re-installs the dependency after removing the
                           external link
  up, update               Updates packages to their latest version based on the specified range

Review your dependencies:
      audit                Checks for known security issues with the installed packages
      licenses             Check licenses in consumed packages
  ls, list                 Print all the versions of packages that are installed, as well as their dependencies, in a
                           tree-structure
      outdated             Check for outdated packages

Run your scripts:
      exec                 Executes a shell command in scope of a project
      run                  Runs a defined package script
      start                Runs an arbitrary command specified in the package's "start" property of its "scripts" object
   t, test                 Runs a package's "test" script, if one was provided

Other:
      pack
      publish              Publishes a package to the registry
      root

Manage your store:
      store add            Adds new packages to the pnpm store directly. Does not modify any projects or files outside
                           the store
      store path           Prints the path to the active store directory
      store prune          Removes unreferenced (extraneous, orphan) packages from the store
      store status         Checks for modified packages in the store

Options:
  -r, --recursive          Run the command for each project in the workspace.
PS E:\use> git clone https://github.com/halo-dev/halo
Cloning into 'halo'...
remote: Enumerating objects: 75321, done.
remote: Counting objects: 100% (2991/2991), done.
remote: Compressing objects: 100% (1410/1410), done.
remote: Total 75321 (delta 1345), reused 2558 (delta 1107), pack-reused 72330
Receiving objects: 100% (75321/75321), 64.47 MiB | 69.00 KiB/s, done.
Resolving deltas: 100% (41471/41471), done.
Updating files: 100% (1472/1472), done.
PS E:\use> cd halo
PS E:\use\halo> cd console
PS E:\use\halo\console> pnpm install
Scope: all 4 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +1385
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Packages are hard linked from the content-addressable store to the virtual store.
  Content-addressable store is at: E:\.pnpm-store\v3
  Virtual store is at:             node_modules/.pnpm

   ╭─────────────────────────────────────────────────────────────────╮
   │                                                                 │
   │                Update available! 8.5.1 → 8.6.7.                 │
   │   Changelog: https://github.com/pnpm/pnpm/releases/tag/v8.6.7   │
   │     Run "corepack prepare pnpm@8.6.7 --activate" to update.     │
   │                                                                 │
   │     Follow @pnpmjs for updates: https://twitter.com/pnpmjs      │
   │                                                                 │
   ╰─────────────────────────────────────────────────────────────────╯

Progress: resolved 1385, reused 1383, downloaded 0, added 1385, done

dependencies:
+ @codemirror/commands 6.1.2
+ @codemirror/lang-css 6.0.1
+ @codemirror/lang-html 6.2.0
+ @codemirror/lang-javascript 6.1.1
+ @codemirror/lang-json 6.0.1
+ @codemirror/language 6.3.1
+ @codemirror/legacy-modes 6.3.0
+ @codemirror/state 6.1.4
+ @codemirror/view 6.5.1
+ @emoji-mart/data 1.0.8
+ @formkit/core 0.17.3
+ @formkit/i18n 0.17.3
+ @formkit/inputs 0.17.3
+ @formkit/themes 0.17.3
+ @formkit/utils 0.17.3
+ @formkit/validation 0.17.3
+ @formkit/vue 0.17.3
+ @halo-dev/api-client 2.7.0 <- packages\api-client
+ @halo-dev/components 1.6.0 <- packages\components
+ @halo-dev/console-shared 2.7.0 <- packages\shared
+ @halo-dev/richtext-editor 0.0.0-alpha.24
+ @tanstack/vue-query 4.29.1
+ @tiptap/extension-character-count 2.0.0-beta.220
+ @tiptap/vue-3 2.0.3
+ @uppy/core 3.2.0
+ @uppy/dashboard 3.4.0
+ @uppy/drag-drop 3.0.2
+ @uppy/file-input 3.0.2
+ @uppy/locales 3.2.1
+ @uppy/progress-bar 3.0.2
+ @uppy/status-bar 3.1.2
+ @uppy/vue 1.0.2
+ @uppy/xhr-upload 3.2.0
+ @vueuse/components 10.2.0
+ @vueuse/core 10.2.0
+ @vueuse/router 10.2.0
+ @vueuse/shared 10.2.0
+ axios 0.27.2
+ codemirror 6.0.1
+ colorjs.io 0.4.3
+ dayjs 1.11.7
+ emoji-mart 5.3.3
+ fastq 1.15.0
+ floating-vue 2.0.0-beta.20
+ fuse.js 6.6.2
+ jsencrypt 3.3.2
+ lodash.clonedeep 4.5.0
+ lodash.debounce 4.0.8
+ lodash.isequal 4.5.0
+ lodash.merge 4.6.2
+ lodash.sortby 4.7.0
+ overlayscrollbars 2.1.1
+ overlayscrollbars-vue 0.5.1
+ path-browserify 1.0.1
+ pinia 2.0.33
+ pretty-bytes 6.0.0
+ qs 6.11.1
+ transliteration 2.3.5
+ vue 3.2.45
+ vue-demi 0.13.11
+ vue-grid-layout 3.0.0-beta1
+ vue-i18n 9.3.0-beta.19
+ vue-router 4.1.6
+ vuedraggable 4.1.0

devDependencies:
+ @changesets/cli 2.25.2
+ @iconify-json/mdi 1.1.50
+ @iconify-json/vscode-icons 1.1.22
+ @intlify/unplugin-vue-i18n 0.9.3
+ @rushstack/eslint-patch 1.3.2
+ @tailwindcss/aspect-ratio 0.4.2
+ @tailwindcss/container-queries 0.1.0
+ @tsconfig/node18 2.0.1
+ @types/jsdom 20.0.1
+ @types/lodash.clonedeep 4.5.7
+ @types/lodash.debounce 4.0.7
+ @types/lodash.isequal 4.5.6
+ @types/lodash.merge 4.6.7
+ @types/node 18.13.0
+ @types/qs 6.9.7
+ @types/randomstring 1.1.8
+ @vitejs/plugin-vue 4.0.0
+ @vitejs/plugin-vue-jsx 3.0.1
+ @vitest/ui 0.25.3
+ @vue/compiler-sfc 3.2.45
+ @vue/eslint-config-prettier 7.1.0
+ @vue/eslint-config-typescript 11.0.3
+ @vue/test-utils 2.3.2
+ @vue/tsconfig 0.4.0
+ autoprefixer 10.4.14
+ c8 7.12.0
+ cypress 10.11.0
+ eslint 8.43.0
+ eslint-plugin-cypress 2.13.3
+ eslint-plugin-vue 9.15.0
+ husky 8.0.3
+ jsdom 20.0.3
+ lint-staged 13.2.2
+ postcss 8.4.21
+ postcss-viewport-height-correction 1.1.1
+ prettier 2.8.8
+ prettier-plugin-tailwindcss 0.1.13
+ randomstring 1.2.3
+ sass 1.60.0
+ start-server-and-test 1.14.0
+ tailwindcss 3.3.0
+ tailwindcss-safe-area 0.2.2
+ tailwindcss-themer 2.0.3
+ typescript 5.0.4
+ unplugin-icons 0.14.15
+ vite 4.0.4
+ vite-compression-plugin 0.0.4
+ vite-plugin-externals 0.6.2
+ vite-plugin-html 3.2.0
+ vite-plugin-pwa 0.13.3
+ vite-plugin-static-copy 0.11.1
+ vitest 0.25.3
+ vue-tsc 1.8.1

. prepare$ cd .. && husky install console/.husky
│ husky - Git hooks installed
└─ Done in 1.6s
Done in 2m 28.5s
PS E:\use\halo\console> pnpm build:packages

> @halo-dev/console@2.7.0 build:packages E:\use\halo\console
> pnpm --filter "./packages/**" build

Scope: 3 of 4 workspace projects
packages/components build$ vite build
│ vite v4.0.4 building for production...
│ transforming...
│ ✓ 172 modules transformed.
│ rendering chunks...
│ computing gzip size...
│ dist/style.css               66.22 kB │ gzip: 10.78 kB
│ dist/halo-components.es.js  188.64 kB │ gzip: 55.83 kB │ map: 398.25 kB
│ dist/halo-components.iife.js  147.59 kB │ gzip: 49.65 kB │ map: 383.44 kB
│ [vite:dts] Start generate declaration files...
│ [vite:dts] Declaration files built in 8139ms.
└─ Done in 51.3s
packages/api-client build$ unbuild
│ i Building @halo-dev/api-client
│ √ Build succeeded for @halo-dev/api-client
│   dist/index.cjs (size: 583 kB, exports: ApiConsoleHaloRunV1alpha1AttachmentApi, ApiConsoleHaloRunV1alpha1Attachment
│   dist/index.mjs (size: 570 kB, exports: ApiConsoleHaloRunV1alpha1AttachmentApi, ApiConsoleHaloRunV1alpha1Attachment
└─ Done in 30.4s
packages/shared build$ vite build
│ vite v4.0.4 building for production...
│ transforming...
│ ✓ 9 modules transformed.
│ rendering chunks...
│ computing gzip size...
│ dist/halo-console-shared.es.js  0.11 kB │ gzip: 0.12 kB │ map: 0.34 kB
│ dist/halo-console-shared.iife.js  0.16 kB │ gzip: 0.15 kB │ map: 0.34 kB
│ [vite:dts] Start generate declaration files...
│ [vite:dts] Declaration files built in 4224ms.
└─ Done in 10.3s
PS E:\use\halo\console> pnpm dev

> @halo-dev/console@2.7.0 dev E:\use\halo\console
> vite --host

<script src="/assets/vue/vue.global.a73fc631.js"></script>                                                    18:30:15
<script src="/assets/vue-router/vue-router.global.a73fc631.js"></script>
<script src="/assets/vue-demi/vue-demi.a73fc631.js"></script>
<script src="/assets/vueuse/vueuse.shared.iife.a73fc631.js"></script>
<script src="/assets/vueuse/vueuse.core.iife.a73fc631.js"></script>
<script src="/assets/vueuse/vueuse.components.iife.a73fc631.js"></script>
<script src="/assets/vueuse/vueuse.router.iife.a73fc631.js"></script>
<script src="/assets/components/halo-components.iife.a73fc631.js"></script>
<script src="/assets/console-shared/halo-console-shared.iife.a73fc631.js"></script>
                                                                                                              18:30:15
  VITE v4.0.4  ready in 9931 ms

  ➜  Local:   http://localhost:3000/console/                                                                  18:30:15
  ➜  Network: http://10.37.114.239:3000/console/                                                              18:30:15
  ➜  Network: http://172.23.160.1:3000/console/                                                               18:30:15
  ➜  press h to show help                                                                                     18:30:15
[vite-plugin-static-copy] Collected 9 items.                                                                  18:30:15

 WARN                                                                                                         18:30:35
A PostCSS plugin did not pass the `from` option to `postcss.parse`. This may cause imported assets to be incorrectly transformed. If you've recently added a PostCSS plugin that raised this warning, please contact the package author to fix the issue.

**Comment by aphantom** [comment]
第一次运行打开就是未知错误
![image](https://github.com/halo-dev/halo/assets/16313864/2f55e688-e922-4296-a491-44d32e3b4591)

**Comment by aphantom** [comment]
不能继续了。

**Comment by ruibaby** [comment]
<img width="1087" alt="image" src="https://github.com/halo-dev/halo/assets/21301288/fa35b2bc-ac0e-4d1d-bbcb-63133a115606">

这一部分的文档在你的浏览器是否无法正常加载？如果是，请告知一下你的浏览器版本，我们尝试复现一下文档显示不全的问题。

**Comment by aphantom** [comment]
8090端口访被拒绝，需要访火墙里面开启吗？

**Comment by guqing** [comment]
> 8090端口访被拒绝，需要访火墙里面开启吗？

首先需要先按照文档启动 halo 服务 然后在运行前端项目也就是用 pnpm运行的部分，然后在访问localhost:8090,如果你的8090访问不到就是你没有运行 halo 的服务端只运行了前端部分
```
# macOS / Linux
./gradlew bootRun --args="--spring.profiles.active=dev"

# Windows
gradlew.bat bootRun --args="--spring.profiles.active=dev,win"
```

**Comment by aphantom** [comment]
哦，似乎你说到点子上了，我马上去试试

**Comment by aphantom** [comment]
这还需要等待？
![image](https://github.com/halo-dev/halo/assets/16313864/0ba6fd23-38db-41da-b122-5c328e0c65a1)

**Reply by guqing** [reply]
> 这还需要等待？
> 
> ![image](https://github.com/halo-dev/halo/assets/16313864/0ba6fd23-38db-41da-b122-5c328e0c65a1)
> 
> 

可以访问到就行 这个状态是gradle任务没有结束所以一直不显示100

**Comment by aphantom** [comment]
可以进后台了，gradle任务一直是 92% EXECUTING [1h 38m 30s] ，有什么问题吗？

**Comment by aphantom** [comment]
感谢@guqing  和  @ruibaby 
每次开启要同时运行 gradlew.bat bootRun --args="--spring.profiles.active=dev,win" 和pnpm dev  这似乎有些麻烦

**Comment by aphantom** [comment]
再次感谢@guqing 和 @ruibaby
